assign-doi-datacite/src/main/java/no/unit/nva/doi/datacite/mdsclient/DataCiteMdsConnection.java

```diff
@@ -59,10 +63,10 @@ public DataCiteMdsConnection(HttpClient httpClient, String host, int port) {
     public HttpResponse<String> postMetadata(String doi, String dataCiteXml) throws IOException,
                                                                                     URISyntaxException,
                                                                                     InterruptedException {
-        Objects.requireNonNull(doi);
-        Objects.requireNonNull(dataCiteXml);
+        Objects.requireNonNull(doi, MISSING_DOI_IDENTIFIER_ARGUMENT);
```

Comment:
The DOI is not parsed at this point, so the fact it is missing is the only relevant information at this point, rather than the structure. Is a DOI-URI also valid?

Should it be parsed now, or later?

This and the following comment applies to all the code.